SPEC.md

```diff
@@ -98,61 +98,114 @@ A build tool is something which reads one or more input artifacts and writes one
 
-When a build tool reads an artifact, it should for each input file:
-1. Compute the git object id of the input file - ${artifact identifier}
-2. Examine the artifact for an embedded GitBOM identifier - ${gitbom idenfitier}
-3. If the artifact contains a GitBOM identifer it construct a record:
-   ```
-   blob⎵${artifact identifier}⎵bom⎵${GitBOM identifier}\n
-   ```
-4. Otherwise construct a record:
-   ```
-   blob⎵${artifact identifier}\n
-   ```
-
-Concatenate all records together and apply lexical ordering by line to create the GitBOM document of the
-output artifact.
+A build tool creating an output artifact must compute a GitBOM Document of each mandatory artifact id type.
+
+For each input artifact the build tool must:
+
+1. Compute the git object id of the input - ${artifact identifier}
+2. Examine the input for an embedded GitBOM Document Identifier - ${gitbom document identifier}
+
+The build tool must persist a GitBOM Document using the ${artifact identifier} and ${gitbom document identifier} for each input.
 
 ### GitBOM Document persistence by a Build Tool
 
-When persisting an output artifact to a file system, the build tool should create a subdirectory of the directory to which the output artifact is being written of the form:
-```.bom/objects/``` and write out the GitBOM document to ```.bom/objects/${gitoid:0:2}/${gitoid:2:}``` where ```${gitoid}```
-is the git object id of the GitBOM document.
+Each build tool should persist all GitBOM Documents it generates.
+
+Generically, GitBOM persistence can be thought of as having an abstract API:
+```
+WriteGitBOM(GitBOM id,GitBOM contents) returns error
+ReadGitBOM(GitBOM id) returns GitBOM Contents
+```
+
+WriteGitBOM should check that GitBOM id matches GitBOM contents and return an error if it does not.
+
+####  GitBOM Document persistence by a Build Tool to its local filesystem
+If a build tool persists GitBOM information to its local filesystem, the build tool should write out the GitBOM Document to ```${GITBOM_DIR}/objects/${GitBOMID:0:2}/${GitBOMID:2:}``` where ```${artifact id}``` is the GitBOM id in lowercase hexidecimal.
 
 ### Build tool persistence of related metadata
 
-When persisting an output artifact to a file system, if the build tool has additional metadata to persist that makes reference to the Artifact Dependency Graph (ADG),
-it should persist that metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```.bom/metadata/${tool}/```.  Filenaming and subdirectory structure below that point is at the discretion of the build tool.
+If the build tool has additional metadata to persist that makes referece to the Artifact Dependency Graph (ADG),
+it should persist that metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```${GITBOM_DIR}/metadata/${tool}/```.  Filenaming and subdirectory structure below that point is at the discretion of the build tool.
```

Comment:
> Is this metadata relevant to the rest of the spec? How does it fit with the rest of the data model? Are we defining a specific format / schema / data model for this metadata? Or is the only relevance that it might reference GitOIDs?

Its relevant as a 'release valve'.  So far pretty much all tool implementors have felt the need for metadata.  Having a common place to put them that is properly partitioned avoids... unfortunate choices.

> 
> How is this different from object files or any other intermediate build artifact? After all, they also contain GitOID references.

Interesting point.  I would argue that the intermediate build artifacts are 'primary data' not 'metadata'.